CMake/kwiver-setup-python.cmake

```diff
@@ -195,6 +195,28 @@ set(PYTHON_ABIFLAGS "${_python_abi_flags}"
 mark_as_advanced(PYTHON_ABIFLAGS)
 
 
+###
+# Python dependencies
+#
+# Add python packages needed to execute the bindings
+# to requirements.txt file
+#
+list(APPEND PYTHON_REQS "numpy>=1.13.0,<=1.19.0" "six>=1.10.0,<=1.13.0")
+
+if(SKBUILD)
+  list(APPEND PYTHON_REQS "scikit-build<=0.11.1")
+endif()
+
+if(KWIVER_ENABLE_PYTORCH)
+  list(APPEND PYTHON_REQS "opencv-python>=3.4.2.17,<=4.0.0"
```

Comment:
This is a tricky package. See this discussion: https://github.com/opencv/opencv-python/issues/467 Do we want opencv-python or opencv-python-headless? For my pip packaged dependencies I settled on not specifying any variant of cv2 in the skbuild.setup requirements. (I did add a graphics, and headless extra_install, but the user has to explicitly pick and install the one they want, otherwise pip installing a package that specifies one or the other my end up breaking somebodies virtualenv). 

Also, does restricting maximum versions prevent a future and potentially compatible version from running? (Warning is ok).

Reply:
Thinking about it, it would be nice if the python requirements spec differentiated between maximum versions where you don't know if it will work or not, and maximum versions that prevent known actually broken code from running.

Reply:
As to the first point, I have no real experience with the distinction between those packages in production, but as Kwiver does use Qt it seems like the same conflicts you experienced could present themselves. With this current setup we could always make it a user configurable option with the default being neither. What are your thoughts there?

These upper limits will restrict future, compatible versions yes. I believe there was a previous discussion on an old PR about how we're maintaining the versions in the requirements.txt and I believe there was at least some small consensus that we would impose upper limits for version ranges to future proof against incompatible packages.

What would be the distinction between those different types of ranges? Maybe I'm missing something, but wouldn't those essentially translate to the same thing in a requirements.txt file? Or did you mean more of an offer two options for version ranges for the packages, one that we know works, and one that may or may not break?

Reply:
The second point is a musing. I don't think there exists any mechanism in pypi to allow it. Just something to think about.

The first point we should make a decision. It might be right to use opencv-python-headless here instead (or make it configurable). I guess what I need clarification on is what these reqs are being used for. Will these be translated to the setuptools install_requires option? (In which case that will influence users who grab the wheel from pypi) Or is this something else?

Reply:
> Also, does restricting maximum versions prevent a future and potentially compatible version from running? (Warning is ok).

I am probably the one who suggested max versions and yes, it's to prevent compatibility issues. A lot of packages are good about deprecating features, API, etc and keeping it as a warning for a reasonable amount of time. But, packages like Numpy (maybe they have changed their ways over the years) have at times been very quick to convert a warning to an error which can cause a lot of immediate panic and scramble.

Either approach requires consistent maintenance but I generally prefer the maintenance coming when someone wants newer features from a package than when a package is left at the latest and breaks one day. That said, if you have a better idea, I am happy to hear it. I'm not deeply familiar with some of the Python best-practices on the topic.